### PR TITLE
Add OSX to travis

### PR DESCRIPTION
**Pull request by libre-man**
Adding OSX to official support.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -17,19 +18,17 @@ cache:
...
+      os: linux
+    - python: "3.6" # 3.6 development branch
+      os: osx
 
 before_install:
   - sudo apt-get install -qq pkg-config fuse
-  - sudo modprobe fuse
-  - sudo chmod 666 /dev/fuse
-  - sudo chown root:$USER /etc/fuse.conf
+  - if [[ "$TRAVIS_OS_NAME" = "osx" ]]; then brew tap caskroom/cask; brew cask install osxfuse; sudo /bin/cp -RfX /usr/local/opt/osxfuse/Library/Filesystems/osxfusefs.fs /Library/Filesystems; sudo chmod +s /Library/Filesystems/osxfusefs.fs/Support/load_osxfusefs; sudo mknod /dev/fuse c 10 229; sudo chmod 666 /dev/fuse; fi
```
Comment (olmokramer) [inline .travis.yml]: Could you put each command on its own line here and on the line below?

**Comment by olmokramer** [comment]
We use them a lot in the backend, yes, but users of the filesystem generally don't run the backend as well. But we don't use them in the filesystem code itself. Another option is to install both 3.5 and 3.6 on travis and run the backend with 3.6.

**Comment by libre-man** [comment]
That is the only option, as the backend simply won't run with <= 3.5. I don't think it is that important to test with 3.5, as long as we simply don't use fstrings we will be fine.

**Comment by libre-man** [comment]
Yes, I know. This PR is very much still in progress.

**Pull request closed without merging**